src/game/client/gameclient.cpp

```diff
@@ -1177,10 +1186,35 @@ void CGameClient::OnNewSnapshot()
 			{
 				m_Snap.m_pGameDataFlag = (const CNetObj_GameDataFlag *)pData;
 				m_Snap.m_GameDataFlagSnapID = Item.m_ID;
+
+				// stats
+				{
+					static int s_FlagCarrierRed = FLAG_ATSTAND;
+					static int s_FlagCarrierBlue = FLAG_ATSTAND;
+					if(s_FlagCarrierRed == FLAG_ATSTAND && m_Snap.m_pGameDataFlag->m_FlagCarrierRed >= 0)
+						OnFlagGrab(TEAM_RED);
+					else if(s_FlagCarrierBlue == FLAG_ATSTAND && m_Snap.m_pGameDataFlag->m_FlagCarrierBlue >= 0)
+						OnFlagGrab(TEAM_BLUE);
+
+					s_FlagCarrierRed = m_Snap.m_pGameDataFlag->m_FlagCarrierRed;
+					s_FlagCarrierBlue = m_Snap.m_pGameDataFlag->m_FlagCarrierBlue;
+				}
 			}
 			else if(Item.m_Type == NETOBJTYPE_FLAG)
 				m_Snap.m_paFlags[Item.m_ID%2] = (const CNetObj_Flag *)pData;
 		}
+
+		// stats
+		for(int i = 0; i < MAX_CLIENTS; i++)
```

Comment:
Should use the client enter/leave messages

Reply:
You're right, the whole Active/WasActive is useless on 0.7